src/CountBase.ts

```diff
@@ -26,72 +34,139 @@ export function createCountBase(disklet: Disklet, databaseName: string, bucketSi
+        if (partitionMetadata === undefined) {
+          partitionMetadata = { length: 0 }
+        }
+        const nextIndex = partitionMetadata.length
 
-      let partitionMetadata = configData.partitions[partition] || {
-        length: 0
-      }
-      const nextIndex = partitionMetadata.length;
-      if (!Number.isNaN(index) || index < 0) {
-        return Promise.reject(new Error('index must be a Number greater than 0'))
-      }
-      if (index > nextIndex) {
-        return Promise.reject(new Error('index is larger than next index in partition'))
-      }
-      if (index === nextIndex) {
-        ++partitionMetadata.length;
-      }
-      let bucketNumber = index / bucketSize;
-      const bucketExists = bucketNumber * bucketSize < nextIndex;
-      let bucketData = [data];
-      if (bucketExists) {
-        return disklet.getText(
-          `${databaseName}/${partition}/${currentBucket}.json`
-        )
-          .then(currentBucketData => {
-            const bucketData = JSON.parse(currentBucketData)
-            bucketData[index] = data
-            return disklet.setText(
-              `${databaseName}/${partition}/${currentBucket}.json`,
-              JSON.stringify(bucketData)
-            ).then(() => {
-              configData.partitions[partition] = partitionMetadata;
+        if (Number.isNaN(index) || index < 0) {
+          return Promise.reject(
+            new Error('index must be a Number greater than 0')
+          )
+        }
+        if (index > nextIndex) {
+          return Promise.reject(
+            new Error('index is larger than next index in partition')
+          )
+        }
+
+        const bucketNumber = Math.floor(index / bucketSize)
+        const bucketIndex = index % bucketSize
+        const bucketExists = bucketIndex !== 0
+        if (index === nextIndex) {
+          ++partitionMetadata.length
+        }
+        if (bucketExists) {
+          return disklet
+            .getText(`${databaseName}/${partition}/${bucketNumber}.json`)
+            .then(currentBucketDataRaw => {
+              const currentBucketData = JSON.parse(currentBucketDataRaw)
+              currentBucketData[bucketIndex] = data
+              return disklet.setText(
+                `${databaseName}/${partition}/${bucketNumber}.json`,
+                JSON.stringify(currentBucketData)
+              )
+            })
+            .then(() => {
+              configData.partitions[partition] = partitionMetadata
               return disklet.setText(
                 `${databaseName}/config.json`,
                 JSON.stringify(configData)
               )
-            }).catch(error => {
-
             })
+            .then(() => null)
+            .catch(error => {
+              return new Error(`Could not insert data. ${error}`)
+            })
+        } else {
+          return disklet
+            .setText(
+              `${databaseName}/${partition}/${bucketNumber}.json`,
```

Comment:
The default partition is / so your path you might have three slashes in a row. Better to have your partition include the / with the name and if there is no partition just have it be an empty string